dist/react-slick.js

```diff
@@ -1106,7 +1102,7 @@ return /******/ (function(modules) { // webpackBootstrap
 	    }
 
 	    if (this.props.beforeChange) {
-	      this.props.beforeChange(currentSlide);
+	      this.props.beforeChange(this.state.currentSlide, currentSlide);
```

Comment:
@vramana I agree. The whole process was more like hacking the component. I didn't expect that will have to submit a PR here :)